plugins/aea-ledger-hwi/aea_ledger_hwi/account.py

```diff
@@ -0,0 +1,408 @@
+        path_construct = PrefixedArray(Byte, Int32ub)
+        path_apdu = path_construct.build(path.to_list())
+
+        if is_eip1559_tx:
+            as_dict = transaction.as_dict()
+            tx = UnsignedDynamicTransaction.from_dict(
+                field_dict=dict(
+                    chain_id=as_dict["chainId"],
+                    nonce=as_dict["nonce"],
+                    max_priority_fee_per_gas=as_dict["maxPriorityFeePerGas"],
+                    max_fee_per_gas=as_dict["maxFeePerGas"],
+                    gas_limit=as_dict.get("gas", 21000),
+                    destination=as_dict["to"],
+                    amount=as_dict["value"],
+                    data=as_dict["data"],
+                    access_list=as_dict.get("accessList", []),
+                )
+            )
+            return (
+                path_apdu
+                + transaction.transaction_type.to_bytes(1, "big")
+                + rlp.encode(tx)
+            )
+
+        as_dict = transaction.as_dict()
+        tx = UnsignedTransaction.from_dict(
+            field_dict=dict(
+                nonce=as_dict["nonce"],
+                gasPrice=as_dict["gasPrice"],
+                gas=as_dict["gas"],
+                to=as_dict["to"],
+                value=as_dict["value"],
+                data=as_dict["data"],
+            )
+        )
+        return path_apdu + rlp.encode(tx)
+
+    def _sign_transaction_on_hwi(
+        self,
+        unsigned_transaction: TypedTransaction,
+        key_index: Optional[int] = None,
+        device_index: Optional[int] = None,
+        chain_id: int = 1,
+    ) -> HWISignedTransaction:
+        """Hardware wallet signed transaction"""
+
+        client = self.get_client(
+            device_index=device_index,
+        )
+
+        is_eip1559_tx = (
+            hasattr(unsigned_transaction, "transaction_type")
+            and unsigned_transaction.transaction_type == 2
+        )
+
+        request_data = self.encode_transaction(
+            transaction=unsigned_transaction,
+            is_eip1559_tx=is_eip1559_tx,
+            key_index=key_index,
+        )
+
+        with reraise_from_hwi_com_error():
+            raw_response = bytes()
+            for idx, each in enumerate(chunk(request_data, 255)):
+                raw_response = client.apdu_exchange(
+                    ins=SignTransactionAPDU.INS,
+                    data=each,
+                    p1=(
+                        SignTransactionAPDU.P1_0 if idx == 0 else SignTransactionAPDU.P1
+                    ),
+                    p2=SignTransactionAPDU.P2,
+                )
+
+        parsed_response = SignedTransactionStruct.parse(raw_response)
+
+        if is_eip1559_tx:
+            return HWISignedTransaction(
+                v=parsed_response.v,
+                r=parsed_response.r,
+                s=parsed_response.s,
+            )
+
+        return HWISignedTransaction(
+            v=(35 + (2 * chain_id)),
```

Comment:
{0,1} + CHAIN_ID * 2 + 35 where {0,1} is the parity of the y value of the curve point for which r is the x-value in the secp256k1 signing process. 
https://eips.ethereum.org/EIPS/eip-155
Please, compare in example https://eips.ethereum.org/EIPS/eip-155Example
https://github.com/ethereum/go-ethereum/issues/19751#issuecomment-504900739
https://www.reddit.com/r/ledgerwallet/comments/1043nhe/ledgers_do_not_sign_ecdsa_properly/

Reply:
Fixed this here [23f443a](https://github.com/valory-xyz/open-aea/pull/583/commits/23f443a0053ced16578bdf560810c0fcc4f10d7a)